junit-platform-commons/src/main/java/org/junit/platform/commons/support/AnnotationSupport.java

```diff
@@ -133,6 +133,23 @@ public static boolean isAnnotated(AnnotatedElement element, Class<? extends Anno
 		return AnnotationUtils.findRepeatableAnnotations(element, annotationType);
 	}
 
+	/**
+	 * Find all {@code public} {@linkplain Field fields} of the supplied class
+	 * or interface that are annotated or <em>meta-annotated</em> with the specified {@code annotationType}.
+	 *
+	 * <p>Consult the Javadoc for {@link Class#getFields()} for details on
+	 * inheritance and ordering.
+	 *
+	 * @param clazz the class or interface in which to find the fields; never {@code null}
+	 * @param annotationType the annotation type to search for; never {@code null}
+	 * @return the list of all such fields found; neither {@code null} nor mutable
+	 * @see Class#getFields()
+	 */
+	public static List<Field> findPublicAnnotatedFields(Class<?> clazz, Class<? extends Annotation> annotationType) {
+
+		return AnnotationUtils.findAnnotatedFields(clazz, annotationType, f -> true);
```

Comment:
That does not limit the results to `public` fields.

Please add a failing test and then fix it. 😉

Reply:
Should we limit the results to public fields or all fields ?

Reply:
Well, the method `findPublicAnnotatedFields(Class<?> clazz, Class<?> fieldType, Class<? extends Annotation> annotationType)` already existed.

So we'll have to leave that existing "public" one in place, but for new methods I think we should only support finding _all_ fields.

Users can then filter the results on their own if they desire.

Plus, we can consider later providing variants that accept a `Predicate<Field> predicate`, but let's avoid that in the current PR.

Reply:
removed public from name